SilverSolver/assignment3/dictionary.py

```diff
@@ -0,0 +1,62 @@
+from typing import Tuple
+
+class Dictionary:
+    """
+    I used some code from here to implement the trie -
+    https://towardsdatascience.com/implementing-a-trie-data-structure-in-python-in-less-than-100-lines-of-code-a877ea23c1a1
+    then modified it for my purposes
+    """
+    
+    def __init__(self, first_char='*', iterable=None):
+        self.char = first_char
+        self.children = []
+        self.word_finished = False
+        if iterable:
+            for word in iterable:
+                self.add(word)
+    
+    def add(self, word: str):
+        node = self
+        for char in word:
+            found_in_child = False
+            # Search for the character in the children of the present `node`
+            for child in node.children:
+                if child.char == char:
+                    # And point the node to the child that contains this char
+                    node = child
+                    found_in_child = True
+                    break
+            # We did not find it so add a new chlid
+            if not found_in_child:
+                new_node = Dictionary(char)
+                node.children.append(new_node)
+                node = new_node
+        node.word_finished = True
+
+    def isPrefixOrWord(self, prefix: str) -> Tuple[bool, bool]:
```

Comment:
Nicer API than what the exercice was recommending!!
I like it that you do not to have to call 2 methods on the Trie, that basically do the same thing .